js/lib/higlass-jupyter.js

```diff
@@ -20,40 +20,165 @@ var HiGlassDisplayModel = widgets.DOMWidgetModel.extend({
 
 // Custom View. Renders the widget model.
 var HiGlassDisplayView = widgets.DOMWidgetView.extend({
-  render: function() {
-    var viewConfig = this.model.get('viewconf');
-    var height = this.model.get('height');
-    var hgOptions = this.model.get('hg_options');
+  render: function render() {
+    this.height = this.model.get('height');
+    this.viewConfig = this.model.get('viewconf');
+    this.authToken = this.model.get('auth_token');
+    this.bounded = this.model.get('bounded');
+    this.defaultTrackOptions = this.model.get('default_track_options');
+    this.darkMode = this.model.get('dark_mode');
+    this.renderer = this.model.get('renderer');
+    this.selectMode = this.model.get('select_mode');
+    this.selectionOnAlt = this.model.get('selection_on_alt');
+    this.options = this.model.get('options');
 
-    var borderColor = hgOptions.isDarkTheme ? '#333333' : '#dddddd'
+    this.hgContainer = document.createElement('div');
+    this.hgDisplay = document.createElement('div');
+    this.hgDisplay.style.border = this.options.theme === 'dark'
+      ? '#333333' : '#dddddd';
+    this.hgDisplay.style.borderRadius = '2px';
 
-    this.hgcontainer = document.createElement('div');
-    this.hgdisplay = document.createElement('div');
-    this.hgdisplay.style.border = '1px solid #dddddd';
-    this.hgdisplay.style.borderRadius = '2px';
+    this.hgContainer.appendChild(this.hgDisplay);
+    this.el.appendChild(this.hgContainer);
 
-    this.hgcontainer.appendChild(this.hgdisplay);
-    this.el.appendChild(this.hgcontainer);
+    this.hg = hglib.viewer(this.hgDisplay, this.viewConfig, this.getOptions());
 
-    if (height) {
-      this.hgdisplay.style.height = height + 'px';
+    window.hgApi = this.hg;
```

Comment:
Why bind to `window`? What happens when there are multiple widget instances?

Reply:
I think this is just for debugging purposes

Reply:
The API is not stored on the `api` property of HiGlass' root DOM container. This container can be retrieved via `document.getElementById(<ID>)` with `<ID>` from `widget.dom_element_id`